src/mono/netcore/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Windows.Mono.cs

```diff
@@ -1,16 +1,39 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
+using System.Runtime.CompilerServices;
+
 
 namespace System.Globalization
 {
     internal partial class GlobalizationMode
     {
-        internal static bool UseNls => true;
-
+        internal static bool UseNls {
+            get {
+                return !Invariant && Interop.Globalization.LoadICU() == 0;
+            }
+        }
         private static bool GetGlobalizationInvariantMode()
         {
-            return GetInvariantSwitchValue();
+            bool invariantEnabled = GetInvariantSwitchValue();
+            if (invariantEnabled)
+                return true;
+
+            LoadICU();
+            return false;
+        }
+
+        // Keep this in a separate method to avoid loading the native lib in invariant mode
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static void LoadICU()
```

Comment:
This method shouldn't be needed in Windows.

Also, we don't want to fail the build in Windows if we didn't find ICU, we want to fallback to use NLS.